gradle-android-command-plugin/src/main/groovy/com/novoda/gradle/command/Monkey.groovy

```diff
@@ -15,12 +16,29 @@ class Monkey extends AdbTask {
 
     @TaskAction
     void exec() {
-        def arguments = ['shell', 'monkey', '-p', packageName, '-v', getEvents()]
+        def arguments = ['shell', 'monkey', '-p', packageName]
+        arguments += getFormattedCategories()
+        arguments += ['-v', getEvents()]
         if (getSeed())
             arguments += ['-s', getSeed()]
         assertDeviceAndRunCommand(arguments)
     }
 
+    def getFormattedCategories() {
+        def formattedCategories = []
+        def categories = getCategories()
+        categories.each {
+            formattedCategories.add("-c " + it)
+        }
+        formattedCategories
+    }
+
+    private getCategories() {
+        if (categories instanceof Closure)
+            categories = categories.call()
```

Comment:
This code seems very... braceless. Not just your addition mind, the whole thing. Scary.